lib/ace/ext/papeeria/tex_completer.coffee

```diff
@@ -121,44 +121,40 @@ define( (require, exports, module) ->
     constructor: ->
       @cachedURL =  ""
       @cache = []
-    getReference: (url, callback) -> 
-      if (url == @cashedURL)
-        return @cashe
-      else 
-        @cachedURL = url
-        json = $.getJSON(url)
-        thisRef = @
-        json.success((data) ->
-          callback(data, thisRef)
-        )
+    processData: (data) => 
+      @cache = data.map((elem) -> 
+                  return {
+                    name: elem.caption
+                    value: elem.caption
+                    score: Number.MAX_VALUE
+                    meta: "ref"
+                  }
+            )
 
-        return @cache
+    getReference: (url) => 
+      if (url != @cashedURL)
+        @cachedURL = url
+        $.getJSON(url).done(@processData)
+        
+      return @cache
 
-  exports.CompletionTools = 
-    TexCompleter: class TexCompleter
+  class TexCompleter
       constructor: ->
-        @r = new ReferenceGetter()
+        @refGetter = new ReferenceGetter()
       getCompletions: (editor, session, pos, prefix, callback) ->
         context = ContextHelper.getContext(session, pos.row)
         token = session.getTokenAt(pos.row, pos.column)
-        console.log(token)
+
         if istype(token, "ref")
-          callback(null, @r.getReference("example.json", (data, r) -> 
-            r.cache = data.map((elem) -> 
-                            return {
-                              name: elem.caption
-                              value: elem.caption
-                              score: Number.MAX_VALUE
-                              meta: "ref"}
-                      )
-            ))
+          callback(null, @refGetter.getReference("example.json"))
```

Comment:
it will not work this way. getJSON sends asynchronous request, the result will be available long after your return from this method. You need to push callback down to the roots.